R/download_data_wrds_compustat.R

```diff
@@ -85,9 +89,44 @@ download_data_wrds_compustat <- function(type, start_date, end_date, additional_
         inv = at / at_lag - 1,
         inv = if_else(at_lag <= 0, NA, inv)
       )
+  }
+
+  if (grepl("compustat_quarterly", type, fixed = TRUE)) {
+    fundq_db <- tbl(con, in_schema("comp", "fundq"))
+
+    compustat <- fundq_db |>
+      filter(
+        indfmt == "INDL" &
+          datafmt == "STD" &
+          consol == "C" &
+          between(datadate, start_date, end_date)
+      ) |>
+      select(
+        gvkey, datadate, rdq, fqtr, fyearq,
+        all_of(additional_columns)
+      ) |>
+      collect()
 
     disconnection_connection(con)
 
-    processed_data
+    compustat <- compustat |>
+      drop_na(fqtr)|>
```

Comment:
Are there observations where this kicks in? Kind of weird tbh. Is fyearq and datadate populated in these cases? Would make more sense to have the main datadate as a filter variable, fqtr seems arbitrary.

Reply:
Agreed! Changed it.